frontend/src/libs/components/app/app.tsx

```diff
@@ -16,27 +36,45 @@ const App: React.FC = () => {
     }
   }, [isRoot, dispatch]);
 
+  useEffect(() => {
+    const checkTokenAndFetchUser = async (): Promise<void> => {
+      if (!(await storage.has(StorageKey.TOKEN))) {
+        navigate(AppRoute.SIGN_IN);
+      }
```

Comment:
we should avoid side effects in components, this check should be in action, also we don't need to navigate to sign in, just do nothing in action if we have no token

navigation to sign-in should be handled by Protected Routes